backend/app/api/routes/dde.py

```diff
@@ -278,14 +280,83 @@ async def extract_from_file(*, session: SessionDep, current_user: CurrentUser, n
         ).model_dump(exclude_unset=True)
     
     chat_completion_response = await ArenaHandler(session, document_data_extractor.owner, chat_completion_request).process_request()
-    extracted_info=chat_completion_response.choices[0].message.content
+    extracted_data=chat_completion_response.choices[0].message.content
+    extracted_data_token = chat_completion_response.choices[0].logprobs.content
     #TODO: handle refusal or case in which content was not correctly done
     # TODO: Improve the prompt to ensure the output is always a valid JSON
-    json_string = extracted_info[extracted_info.find('{'):extracted_info.rfind('}')+1]
-    extracted_data = {k: v for k, v in json.loads(json_string).items() if k not in ('source', 'year')}   
-    logprob_data = extract_logprobs_from_response(chat_completion_response, extracted_data)
-    return {'extracted_info': json.loads(json_string), 'logprob_data': logprob_data}
+    json_string = extracted_data[extracted_data.find('{'):extracted_data.rfind('}')+1]
+    keys = list(pydantic_reponse.__fields__.keys())
+    value_indices = extract_tokens_indices_for_each_key(keys, extracted_data_token)
+    logprobs = extract_logprobs_from_indices(value_indices, extracted_data_token)
+    return {'extracted_data': json.loads(json_string), 'logprobs': logprobs}
+
+class Token(TypedDict):
+    token: str
+    
+def extract_tokens_indices_for_each_key(keys: list[str], token_list:list[Token]) -> dict[str, list[int]]:
+    """
+    Extracts the indices of tokens corresponding to extracted data related to a list of specified keys.
+
+    The extraction criteria are based on the following:
+    - The function looks for tokens that match the specified keys.
+    - It saves the indices of the tokens that correspond to the values extracted by the model for each key.
+    - Tokens' indices are saved if they follow the pattern '":' or '":"'.
+    - It stops saving indices if it encounters a token that indicates the start of a new key or the end of the object.
 
+    Args:
+        keys (list[str]): A list of keys for which to find corresponding token indices.
+        token_list (list[Token]): A list of Token objects, each containing a token.
+
+    Returns:
+        dict[str, list[int]]: A dictionary mapping each key to the corresponding indices of the tokens representing the values extracted by the model.
+    """
+    value_indices = {key: [] for key in keys}
+    current_key = ""
+    matched_key = None
+    remaining_keys = keys.copy()
+    saving_indices = False 
+    for i, token_object in enumerate(token_list):
+        token = token_object.token
+        if matched_key is not None:
+            if saving_indices:
+                if token == '","' or token == ',"':
```

Comment:
Are we sure no other token contains `,` ?
What if the value is a string containing the `,` character?